Fix: logic for `ee_manager_visible` is falsey

wielinde opened the pull request:
`allows_to?` and `show_banners?` should behave differently if the
configuration is set to false.

wielinde committed:
Fix: logic for `ee_manager_visible` is falsey

`allows_to?` and `show_banners?` should behave differently if the
configuration is set to false.

wielinde committed:
Fix: undo accidental changing the default of `ee_manager_visible`

oliverguenther commented on app/services/authorization/enterprise_service.rb:
```diff
@@ -37,7 +37,7 @@ def initialize(token)
   def call(action)
     allowed =
-      if token.nil? || token.token_object.nil? || token.expired?
+      if OpenProject::Configuration.ee_manager_visible? && (token.nil? || token.token_object.nil? || token.expired?)
```
This shouldn't influence the service result. It should still return the correct check for a specific feature.

wielinde replied:
@oliverguenther OK, what do you propose? The purpose is clear, right? I want to be able to ask one thing and know if I am allowed to use a feature or not.

oliverguenther replied:
Don't we have two uses cases? You either want to find out whether a given feature is available or whether you want to show banners in case you don't have it.

That would mean that `Token.allows_to? ...` should return the same value regardless of the configuration value.

wielinde committed:
Untangle logic.

`ee_manager_visible` only effects visibility of banners advertising EE.
It does not effect `allows_to?` checks on the token.

wielinde committed:
Merge branch 'dev' into fix/central-configuration-for-ee
